src/OpenTelemetry.Api/Context/Propagation/BaggagePropagator.cs

```diff
@@ -94,7 +94,7 @@ public override void Inject<T>(PropagationContext context, T carrier, Action<T,
                     continue;
                 }
 
-                baggage.Append(WebUtility.UrlEncode(item.Key)).Append('=').Append(WebUtility.UrlEncode(item.Value)).Append(',');
+                baggage.Append(W3CBaggageValueEncoder.Encode(item.Key)).Append('=').Append(W3CBaggageValueEncoder.Encode(item.Value)).Append(',');
```

Comment:
Baggage item keys should meet `token` requirement from the [specification](https://w3c.github.io/baggage/#key), and not be encoded, I can address that in a separate PR.